checks/raw/code_review.go

```diff
@@ -15,21 +15,210 @@
 	// Look at the latest commits.
 	commits, err := c.ListCommits()
 	if err != nil {
 		return checker.CodeReviewData{}, fmt.Errorf("%w", err)
 	}
 
+	changesets := getChangesets(commits, dl)
+
+	if err != nil {
+		return checker.CodeReviewData{}, fmt.Errorf("%w", err)
+	}
+
 	return checker.CodeReviewData{
-		DefaultBranchCommits: commits,
+		DefaultBranchChangesets: changesets,
 	}, nil
 }
+
+func getGithubRevisionID(c *clients.Commit, dl checker.DetailLogger) string {
+	mr := c.AssociatedMergeRequest
+	if !c.AssociatedMergeRequest.MergedAt.IsZero() {
+		dl.Info(&checker.LogMessage{
+			Text: fmt.Sprintf("commit %s was reviewed on github #%d approved merge request",
+				c.SHA, c.AssociatedMergeRequest.Number),
+		})
+		if mr.Number != 0 {
+			return strconv.Itoa(mr.Number)
+		}
+	}
+	return ""
+}
+
+func getProwRevisionID(c *clients.Commit, dl checker.DetailLogger) string {
+	mr := c.AssociatedMergeRequest
+	if !c.AssociatedMergeRequest.MergedAt.IsZero() {
+		for _, l := range c.AssociatedMergeRequest.Labels {
+			if l.Name == "lgtm" || l.Name == "approved" {
+				dl.Info(&checker.LogMessage{
+					Text: fmt.Sprintf("commit %s was reviewed on prow #%d approved merge request",
+						c.SHA, c.AssociatedMergeRequest.Number),
+				})
+				if mr.Number != 0 {
+					return strconv.Itoa(mr.Number)
+				}
+			}
+		}
+	}
+
+	return ""
+}
+
+func getGerritRevisionID(c *clients.Commit, dl checker.DetailLogger) string {
+	m := c.Message
+	if strings.Contains(m, "Reviewed-on:") &&
+		strings.Contains(m, "Reviewed-by:") {
+		dl.Info(&checker.LogMessage{
+			Text: fmt.Sprintf("commit %s was reviewed on gerrit", c.SHA),
+		})
+		return c.SHA
+	}
+	return ""
+}
+
+// Given m, a commit message, find the Phabricator revision ID in it.
+func getPhabricatorRevisionID(c *clients.Commit, dl checker.DetailLogger) string {
+	m := c.Message
+	p, err := regexp.Compile(`Differential Revision:\s*(\w+)`)
+	if err != nil {
+		dl.Debug((&checker.LogMessage{Text: "phabricator revisionID regex compile failed"}))
+		return ""
+	}
+	match := p.FindStringSubmatch(m)
+	if match == nil || len(match) < 2 {
+		return ""
+	}
+
+	dl.Info(&checker.LogMessage{
+		Text: fmt.Sprintf("commit %s was reviewed on phabricator revision %s", c.SHA, match[1]),
+	})
+
+	return match[1]
+}
+
+// Given m, a commit message, find the piper revision ID in it.
+func getPiperRevisionID(c *clients.Commit, dl checker.DetailLogger) string {
+	m := c.Message
+	matchPiperRevID, err := regexp.Compile(`PiperOrigin-RevId:\s*(\d{3,})`)
+	if err != nil {
+		dl.Debug((&checker.LogMessage{Text: "piper regex compile failed"}))
+		return ""
+	}
+
+	match := matchPiperRevID.FindStringSubmatch(m)
+
+	if match == nil || len(match) < 2 {
+		return ""
+	}
+
+	dl.Info(&checker.LogMessage{
+		Text: fmt.Sprintf("commit %s reviewed through piper revision %s", c.SHA, match[1]),
+	})
+
+	return match[1]
+}
+
+func getCommitRevisionByPlatform(
+	c *clients.Commit,
+	dl checker.DetailLogger,
+) (string, string, error) {
```

Comment:
It might be helpful to define a struct which holds both `reviewPlatform` and `revisionID`. It'll improve the return signature of this function and also can be used to define helper fns like: `newlyDefinedStruct.isEqual()`